### ISSUE TITLE
Code Generation throws "error: method asList in class Arrays cannot be applied to given types"

### ISSUE THREAD

**Issue by Parilar** [apollographql/apollo-android#2265]
**Summary**
When building the project and the codes gets generated the following two error came up

`error: method asList in class Arrays cannot be applied to given types;
    private Input<List<ProfileWhereUniqueInput>> receivers = Input.fromNullable(Arrays.<ProfileWhereUniqueInput>asList("null"));`

`error: incompatible types: Input<String> cannot be converted to Input<ConversationWhereUniqueInput>
    private Input<ConversationWhereUniqueInput> conversation = Input.fromNullable("null");`

**Description**
schema
```
input MessageInput {
    conversation: ConversationWhereUniqueInput
    receivers: [ProfileWhereUniqueInput!]
    text: String!
}

input ConversationWhereUniqueInput {
    id: String
}

input ProfileWhereUniqueInput {
    id: String
}
```

**Version**
2.0.3

**Comment by martinbonnin** [comment]
That sounds like your schema contains a default value of `"null"` where it should be something like `"[]"`. Can you share the schema.json file ?

**Comment by Parilar** [comment]
Okay, we had default: null in one of our Input Types. The solution was to remove it, now i have to set it manually to null

**Comment by martinbonnin** [comment]
Interesting. How do you define the defaultValue ? Do you have quotes in your schema.json ? 

If yes, can you try removing them? 

```
// replace
"defaultValue": "null"
// with
"defaultValue": null
```

**Comment by Parilar** [comment]
```
t.list.field('receivers', {
            type: 'ProfileWhereUniqueInput',
            nullable: true,
            default: null,
        });
```

This was the old code. Removing the default: null fixed the problem

**Comment by mazzaker** [comment]
schema.graphql:
```
input MessageInput {
  receivers: [ProfileWhereUniqueInput!] = null
  conversation: ConversationWhereUniqueInput = null
  text: String!
}
```

schema.json get with apollo cli:
```
{
        "kind": "INPUT_OBJECT",
        "name": "MessageInput",
        "description": null,
        "fields": null,
        "inputFields": [
          {
            "name": "receivers",
            "description": null,
            "type": {
              "kind": "LIST",
              "name": null,
              "ofType": {
                "kind": "NON_NULL",
                "name": null,
                "ofType": {
                  "kind": "INPUT_OBJECT",
                  "name": "ProfileWhereUniqueInput",
                  "ofType": null
                }
              }
            },
            "defaultValue": "null"
          },
          {
            "name": "conversation",
            "description": null,
            "type": {
              "kind": "INPUT_OBJECT",
              "name": "ConversationWhereUniqueInput",
              "ofType": null
            },
            "defaultValue": "null"
          },
          {
            "name": "text",
            "description": null,
            "type": {
              "kind": "NON_NULL",
              "name": null,
              "ofType": {
                "kind": "SCALAR",
                "name": "String",
                "ofType": null
              }
            },
            "defaultValue": null
          }
        ],
        "interfaces": null,
        "enumValues": null,
        "possibleTypes": null
      },
```
There are quotes in the schema.json, maybe the bug is in apollo cli?

**Comment by mazzaker** [comment]
Find that GraphQL stringify defaultValue so this is no bug. Setting required: false for the types results in defaultValue: null

**Comment by sav007** [comment]
@martinbonnin can we close this one?

**Comment by martinbonnin** [comment]
The issue is still there. While #2275 technically did fix this specific issue, I did not feel confident enough with the Java codegen to merge it. Let me try again tomorrow.
